packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -123,53 +127,123 @@ describe('StripeHelper', () => {
+      assert(mockRedis.get.calledOnce);
 
-    assert.lengthOf(await stripeHelper.allPlans(), 3);
-    assert(mockRedis.get.calledTwice);
-    assert(mockRedis.set.calledOnce);
-    assert(stripeHelper.stripe.plans.list.calledOnce);
+      assert.lengthOf(await stripeHelper.allProducts(), 3);
+      assert(mockRedis.get.calledTwice);
+      assert(mockRedis.set.calledOnce);
+      assert(stripeHelper.stripe.products.list.calledOnce);
 
-    assert.deepEqual(
-      await stripeHelper.allPlans(),
-      JSON.parse(await mockRedis.get('listPlans'))
-    );
+      assert.deepEqual(
+        await stripeHelper.allProducts(),
+        JSON.parse(await mockRedis.get('listProducts'))
+      );
+    });
   });
 
-  it('pulls a list of products and caches it', async () => {
-    assert.lengthOf(await stripeHelper.allProducts(), 3);
-    assert(mockRedis.get.calledOnce);
+  describe('verifyPlanUpgradeForSubscription', () => {
+    it('does nothing for valid upgrade', async () => {
+      assert.isUndefined(
+        await stripeHelper.verifyPlanUpgradeForSubscription(
+          'plan_G93lTs8hfK7NNG',
+          'plan_G93mMKnIFCjZek'
+        )
+      );
+    });
 
-    assert.lengthOf(await stripeHelper.allProducts(), 3);
-    assert(mockRedis.get.calledTwice);
-    assert(mockRedis.set.calledOnce);
-    assert(stripeHelper.stripe.products.list.calledOnce);
+    describe('when the upgrade is invalid', () => {
+      it('throws an invalidPlanUpgrade error', async () => {
+        return stripeHelper
+          .verifyPlanUpgradeForSubscription(
+            'plan_G93lTs8hfK7NNG',
+            'plan_F4G9jB3x5i6Dpj'
+          )
+          .then(
+            () => Promise.reject(new Error('Method expected to reject')),
+            err => {
+              assert.equal(err.errno, error.ERRNO.INVALID_PLAN_UPGRADE);
+            }
+          );
+      });
+    });
 
-    assert.deepEqual(
-      await stripeHelper.allProducts(),
-      JSON.parse(await mockRedis.get('listProducts'))
-    );
+    describe('when the current plan specified does not exist', () => {
+      it('thows an unknownSubscriptionPlan error', async () => {
+        return stripeHelper
+          .verifyPlanUpgradeForSubscription('plan_bad', 'plan_F4G9jB3x5i6Dpj')
+          .then(
+            () => Promise.reject(new Error('Method expected to reject')),
+            err => {
+              assert.equal(err.errno, error.ERRNO.UNKNOWN_SUBSCRIPTION_PLAN);
+            }
+          );
+      });
+    });
+
+    describe('when the new plan specified does not exist', () => {
+      it('thows an unknownSubscriptionPlan error', async () => {
+        return stripeHelper
+          .verifyPlanUpgradeForSubscription('plan_F4G9jB3x5i6Dpj', 'plan_bad')
+          .then(
+            () => Promise.reject(new Error('Method expected to reject')),
+            err => {
+              assert.equal(err.errno, error.ERRNO.UNKNOWN_SUBSCRIPTION_PLAN);
+            }
+          );
+      });
```

Comment:
Was this approach easier to reason about then doing something like:
```javascript
try {
    await stripeHelper.verifyPlanUpgradeForSubscription('plan_F4G9jB3x5i6Dpj', 'plan_bad');
    assert.fail('Method expected to reject');
} catch (err) {
    assert.equal(err.errno, error.ERRNO.UNKNOWN_SUBSCRIPTION_PLAN);
}
```

Reply:
I just saw it done both ways in the code base, and it just made more sense to me because the structure clearly shows both of the possible outcomes.